src/index.ts

```diff
@@ -0,0 +1,55 @@
+import Web3 from 'web3';
+import { StorageProvider, IpfsStorageProvider, IpfsStorageProviderSettings } from './storage';
+import ContractsManager from './contracts';
+import { ProofPointsRepo } from './proofPoints';
+
+interface ProvenanceSettings {
+  web3: Web3;
+  storageSettings: IpfsStorageProviderSettings;
+  proofPointStorageAddress: string;
+}
+
+class Provenance {
+  web3: Web3;
+  contracts: ContractsManager;
+  storage: StorageProvider;
+  proofPoint: ProofPointsRepo;
+
+  constructor(settings: ProvenanceSettings) {
+    if (typeof settings === 'undefined') {
+      // eslint-disable-next-line no-param-reassign
+      settings = {
+        web3: undefined,
+        storageSettings: undefined,
+        proofPointStorageAddress: undefined
+      }
+    }
+
+    if (typeof settings.web3 === 'undefined') {
+      throw new Error('web3 must be defined');
```

Comment:
Out of interest, are there any more specific error types that would be more appropriate?